client/state/data-layer/wpcom/me/settings/test/index.js

```diff
@@ -0,0 +1,166 @@
+				const action = { type: 'DUMMY' };
+
+				settingsModule.requestUserSettings( { dispatch }, action, next );
+
+				expect( next ).to.have.been.calledWith( action );
+			} );
+		} );
+
+		describe( '#storeFetchedUserSettings', () => {
+			it( 'should dispatch user settings update', () => {
+				const action = { type: 'DUMMY' };
+
+				settingsModule.storeFetchedUserSettings( { dispatch }, action, next, {
+					description: 'foo &amp; bar',
+					language: 'qix',
+				} );
+
+				expect( dispatch ).to.have.been.calledOnce;
+				expect( dispatch ).to.have.been.calledWith( updateUserSettings( {
+					description: 'foo & bar',
+					language: 'qix',
+				} ) );
+			} );
+		} );
+	} );
+
+	describe( 'user settings save', () => {
+		describe( '#saveUserSettings', () => {
+			it( 'should dispatch POST request to me/settings using unsavedSettings from state', () => {
+				const getState = () => ( {
+					userSettings: {
+						settings: { foo: 'bar' },
+						unsavedSettings: { foo: 'baz' }
+					}
+				} );
+				const action = { type: 'DUMMY' };
+
+				settingsModule.saveUserSettings( { dispatch, getState }, action, next );
+
+				expect( dispatch ).to.have.been.calledOnce;
+				expect( dispatch ).to.have.been.calledWith( http( {
+					apiVersion: '1.1',
+					method: 'POST',
+					path: '/me/settings',
+					body: { foo: 'baz' }
+				}, action ) );
+				expect( next ).to.have.been.calledWith( action );
+			} );
+
+			it( 'should dispatch POST request to me/settings using explicit settingsOverride', () => {
+				const getState = () => ( {} );
+				const action = {
+					type: 'DUMMY',
+					settingsOverride: { foo: 'baz' }
+				};
+
+				settingsModule.saveUserSettings( { dispatch, getState }, action, next );
+
+				expect( dispatch ).to.have.been.calledOnce;
+				expect( dispatch ).to.have.been.calledWith( http( {
+					apiVersion: '1.1',
+					method: 'POST',
+					path: '/me/settings',
+					body: { foo: 'baz' }
+				}, action ) );
+				expect( next ).to.have.been.calledWith( action );
+			} );
+
+			it( 'should not dispatch any HTTP request when there are no unsaved settings', () => {
+				const getState = () => ( {
+					userSettings: {
+						settings: {},
+						unsavedSettings: {}
+					}
+				} );
+				const action = { type: 'DUMMY' };
+
+				settingsModule.saveUserSettings( { dispatch, getState }, action, next );
+
+				expect( dispatch ).to.not.have.been.called;
+				expect( next ).to.have.been.calledWith( action );
+			} );
+		} );
+
+		describe( '#finishUserSettingsSave', () => {
+			it( 'should dispatch user settings update and clear all unsaved settings on full save', () => {
+				const action = { type: 'DUMMY' };
+
+				settingsModule.finishUserSettingsSave( { dispatch }, action, next, {
+					description: 'foo &amp; bar',
+				} );
+
+				expect( dispatch ).to.have.been.calledTwice;
+				expect( dispatch ).to.have.been.calledWith( updateUserSettings( {
+					description: 'foo & bar'
```

Comment:
should we be testing the entity decoding _and_ the save behavior in the same test? it seems like if some error with the decoded cropped up it would mistakenly fail this test.

maybe if we have a test for the decoding we could consider removing it from the other tests too?